### PR TITLE
win_chocolatey: Clean up parameter handling

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

win_chocolatey

##### ANSIBLE VERSION

v2.3

##### SUMMARY

Changes include:

- Use Get-AnsibleParam with -type/-validateset
- Replace $result PSObject with normal hash
- Deprecate 'upgrade' parameter by using state=latest

### CODE REVIEW

**Comment by dagwieers** [comment]
ready_for_review

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
Looks good shipit

File: lib/ansible/modules/windows/win_chocolatey.ps1 [hunk 1]
```diff
@@ -60,60 +63,65 @@ Function Chocolatey-Install-Upgrade
...
         $script:executable = "C:\ProgramData\chocolatey\bin\choco.exe"
+
     }
     else
     {
+
         $script:executable = "choco.exe"
 
-        if ([Version](choco --version) -lt [Version]'0.9.9')
+        if ([Version](choco --version) -lt [Version]'0.9.9')xi
```
Comment (nitzmahone) [inline lib/ansible/modules/windows/win_chocolatey.ps1]: line garbage?

File: lib/ansible/modules/windows/win_chocolatey.ps1 [hunk 2]
```diff
@@ -60,60 +63,65 @@ Function Chocolatey-Install-Upgrade
...
-    } 
-    
-    If ("$results" -match "$package .* (\d+) packages installed.")
+        $result.choco_error_cmd = $cmd
+        $result.choco_error_log = $output
+
+        Throw "Error checking installation status for $package"
+    }
+
+    If ("$results" -match "$package .* (\d+) packages installed.")xi
```
Comment (nitzmahone) [inline lib/ansible/modules/windows/win_chocolatey.ps1]: garbage?

File: lib/ansible/modules/windows/win_chocolatey.ps1 [hunk 3]
```diff
@@ -135,7 +143,7 @@ Function Choco-Upgrade
         [bool]$ignoredependencies
     )
 
-    if (-not (Choco-IsInstalled $package))
+    if (-not (Choco-IsInstalled $package))xi
```
Comment (nitzmahone) [inline lib/ansible/modules/windows/win_chocolatey.ps1]: garbage?

**Review by nitzmahone — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_chocolatey.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_chocolatey.ps1]: Damn, these are VIM commands :-)
Part of cleaning up the formatting. What's weird is that CI didn't trip on this ?
Fixed.

File: lib/ansible/modules/windows/win_chocolatey.ps1 (hunk 3 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_chocolatey.ps1]: Fixed.

File: lib/ansible/modules/windows/win_chocolatey.ps1 (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_chocolatey.ps1]: Fixed.

**Comment by dagwieers** [comment]
ready_for_review

**Comment by gundalow** [comment]
Merged, one shipit from resmo and nitzmahone's comments have been addressed

**Pull request merged**